internal/controller/servermaintenance_controller.go

```diff
@@ -447,9 +473,27 @@ func (r *ServerMaintenanceReconciler) enqueueMaintenanceByClaimRefs() handler.Ev
 		if _, ok := annotations[metalv1alpha1.ServerMaintenanceNeededLabelKey]; !ok {
 			return req
 		}
+		if claim.Spec.ServerRef == nil {
+			log.V(1).Info("ServerClaim has no ServerRef, skipping")
+			return req
+		}
+		server := &metalv1alpha1.Server{}
+		err := r.Get(ctx, client.ObjectKey{
+			Name: claim.Spec.ServerRef.Name,
+		}, server)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.V(1).Info("Server not found for ServerClaim, skipping", "ServerClaim", claim.Name)
+				return req
+			}
+			log.Error(err, "failed to get server for ServerClaim", "ServerClaim", claim.Name)
+			return nil
+		}
 
 		maintenanceList := &metalv1alpha1.ServerMaintenanceList{}
-		if err := r.List(ctx, maintenanceList); err != nil {
+		if err := r.List(ctx, maintenanceList, client.MatchingFields{
+			ServerMaintenanceServerRefIndex: server.Name,
```

Comment:
In the index we can check for `claim.Spec.ServerRef.Name`. This would avoid fetching the `Server`.

Reply:
yeah true, I was even using it to get the server :D